packages/api/lambdas/internal-reconciliation-report.js

```diff
@@ -0,0 +1,278 @@
+'use strict';
+
+const cloneDeep = require('lodash/cloneDeep');
+const pick = require('lodash/pick');
+const sortBy = require('lodash/sortBy');
+const isEqual = require('lodash/isEqual');
+const union = require('lodash/union');
+const omit = require('lodash/omit');
+const moment = require('moment');
+const pLimit = require('p-limit');
+const Logger = require('@cumulus/logger');
+const { constructCollectionId } = require('@cumulus/message/Collections');
+const { s3 } = require('@cumulus/aws-client/services');
+const { ESSearchQueue } = require('../es/esSearchQueue');
+const { Collection, Granule } = require('../models');
+const {
+  convertToCollectionSearchParams,
+  convertToGranuleSearchParams,
+  initialReportHeader,
+} = require('../lib/reconciliationReport');
+const log = new Logger({ sender: '@api/lambdas/internal-reconciliation-report' });
+
+/**
+ * Compare the collection holdings in Elasticsearch with Database
+ *
+ * @param {Object} recReportParams - lambda's input filtering parameters to
+ *                                   narrow limit of report.
+ * @returns {Promise<Object>} an object with the okCount, onlyInEs, onlyInDb
+ * and withConfilcts
+ */
+async function reconciliationReportForCollections(recReportParams) {
```

Comment:
@jennyhliu: This was a thought I had doing the review, but forgot to put it in here. Can we rename the major functions in here to be internal report specific?  I think it will confuse me when there's a reconciliationReportForCollections in two files.  what about internalRRForCollections or internalReportForCollections or internalRecReportForCollections?  (I like the second best)

and the same thing for reconciliationReportForGranules => internalReportForGranules or similar?
What do you think?

Reply:
Yeah, I should make these functions more specific.